AK/URLParser.cpp

```diff
@@ -70,31 +70,30 @@ String urldecode(const StringView& input)
     return builder.to_string();
 }
 
-static inline bool in_c0_control_set(u32 c)
+String urlencode(const StringView& input, URLEncodeMode mode)
 {
-    return c <= 0x1f || c > '~';
-}
-
-static inline bool in_fragment_set(u32 c)
-{
-    return in_c0_control_set(c) || c == ' ' || c == '"' || c == '<' || c == '>' || c == '`';
-}
+    switch (mode) {
+    case URLEncodeMode::PreserveSpecialCharacters:
+        return urlencode(input, "/?:@-._~!$&'()*+,;="); // https://www.ietf.org/rfc/rfc2396.txt section 2.2 and 2.3, "reserved characters" and "unreserved characters".
```

Comment:
Please use https://url.spec.whatwg.org/ as a reference for URL encoding and decoding :)

Reply:
Oh my, a fancy specification that's not a text file written a few months after I was born!

Reply:
That spec seems to be oddly silent about what to do when serialising query strings (it seems to assume that the query string is sourced from a valid url, and so the query string is definitely encoded correctly?)
I don't think I have the patience to read this spec fully, I'll just close this PR for now and revisit it later